aiida_quantumespresso/calculations/__init__.py

```diff
@@ -244,6 +312,55 @@ def prepare_for_submission(self, folder):
 
         return calcinfo
 
+    def _add_parallelization_flags_to_cmdline_params(self, cmdline_params):
+        """Get the command line parameters with added parallelization flags.
+
+        Adds the parallelization flags to the given `cmdline_params` and
+        returns the updated list.
+
+        Raises an `InputValidationError` if multiple aliases to the same
+        flag are given in `cmdline_params`, or the same flag is given
+        both in `cmdline_params` and the explicit `parallelization`
+        input.
+        """
+        cmdline_params_res = copy.deepcopy(cmdline_params)
+        # The `cmdline_params_normalized` are used only here to check
+        # for existing parallelization flags.
+        cmdline_params_normalized = []
+        for param in cmdline_params:
+            cmdline_params_normalized.extend(param.split())
+
+        if 'parallelization' in self.inputs:
+            parallelization_dict = self.inputs.parallelization.get_dict()
+        else:
+            parallelization_dict = {}
+        # To make the order of flags consistent and "nice", we use the
+        # ordering from the flag definition.
+        for flag_name in self._ENABLED_PARALLELIZATION_FLAGS:
+            all_aliases = list(self._PARALLELIZATION_FLAG_ALIASES[flag_name]) + [flag_name]
+            aliases_in_cmdline = [alias for alias in all_aliases if f'-{alias}' in cmdline_params_normalized]
+            if aliases_in_cmdline:
+                if len(aliases_in_cmdline) > 1:
+                    raise exceptions.InputValidationError(
```

Comment:
I prefer `ValueError`s here personally, because `InputValidationError` does not inherit from it and so you will have to catch it explicitly. Same for line 349

Reply:
I was following the example of the surrounding code - should I change those also?

Reply:
Maybe let's keep it the same then for now for consistency.